Automattic/wp-calypso
# Ensure stacked comparison page is logging events correctly

This PR primarily ensures that tracking events for email comparison views are being sent correctly, especially for views of the `EmailProvidersStackedComparison` component. As part of this, we ensure that the event `calypso_email_providers_comparison_page_view` is sent for all comparison views, including the stacked email comparison view.

#### Changes proposed in this Pull Request

* Adds optionality to some props of `EmailProvidersStackedComparison` to ensure that earlier invocations are now compliant
* Rectifies the import alias of the stacked comparison view from `EmailProvidersComparisonStacked` to `EmailProvidersStackedComparison`
* Exchanges the `PageViewTracker` for a `TrackComponentView` for component `EmailProvidersStackedComparison`
* Adds a new `EmailProvidersStackedComparisonPage` component to host the comparison view and track page views
* Adds page view tracking for the inbox home

#### Test Preparation



* This PR works on and establishes track events, so setup up your environment to discover/monitor track events as they are fired. There are a number of FG guides that show how to do this: PCYsg-rdg-p2
* Get a site and a single domain without an email subscription if you don't have one. We are essentially testing the email comparison page and that requires a domain without an email subscription.


#### Testing instructions



* Checkout the branch or use [calypso-live](https://github.com/Automattic/wp-calypso/pull/60631#issuecomment-1024468645) 
* Navigate to [ **Upgrades** -> **Emails** ]. 
* Confirm that you are presented with the stacked email comparison page.
* Confirm that `calypso_page_view` and `calypso_email_providers_comparison_page_view` events are triggered.
* Navigate to [ **Upgrades** -> **Domains**  ].  Click on **+Add** in the `Email` column. 
* Confirm that you are presented with the stacked email comparison page.
* Confirm that `calypso_page_view` and `calypso_email_providers_comparison_page_view` events are triggered.
* Navigate to [ **Inbox** ]. You should be presented with the stacked email comparison page.
* Confirm that you are presented with the stacked email comparison page.
* Confirm that `calypso_page_view` and `calypso_email_providers_comparison_page_view` events are triggered.
